.github/ISSUE_TEMPLATE/release-checklist.md

```diff
@@ -0,0 +1,72 @@
+---
+name: Release a new version
+about: Start the process for a new release
+---
+
+# Release process
+
+This project uses [cargo-release][cargo-release] in order to prepare new releases, tag and sign relevant git commit, and publish the resulting artifacts to [crates.io][crates-io].
+The release process follows the usual PR-and-review flow, allowing an external reviewer to have a final check before publishing.
+
+## Requirements
+
+This guide requires:
+
+ * a web browser (and network connectivity)
+ * `git`
+ * GPG setup and personal key for signing
+ * `cargo` (suggested: latest stable toolchain from [rustup][rustup])
+ * `cargo-release` (suggested: `cargo install -f cargo-release`)
+ * A verified account on crates.io
+ * Write access to this GitHub project
+ * Upload access to this GitHub project
+
+## Release checklist
+
+These steps show how to release version `x.y.z` on the `origin` remote (this can be checked via `git remote -av`).
+Push access to the upstream repository is required in order to publish the new tag and the PR branch.
+
+- [ ] make sure the project is clean and prepare the environment:
+  - [ ] `cargo test`
+  - [ ] `cargo clean`
+  - [ ] `git clean -fd`
+  - [ ] `export RELEASE_VER=x.y.z`
+  - [ ] `export UPSTREAM_REMOTE=origin`
+
+:warning:: `UPSTREAM_REMOTE` should reference the locally configured remote that points to the upstream git repository.
+
+- [ ] create release commits on a dedicated branch and tag it:
+  - [ ] `git checkout -b release-${RELEASE_VER}`
+  - [ ] `cargo release` (and confirm the version when prompted)
+
+- [ ] open a PR for this release
+  - [ ] `git push ${UPSTREAM_REMOTE} release-${RELEASE_VER}`
```

Comment:
Why push the branch to upstream instead of a fork?

Reply:
I think a fork or upstream would work. Either way there is a step later for deleting the release PR branch.

Reply:
Both would work. The release captain needs to push the tag to the upstream remote anyway, so we just use that same remote for the branch too.